flowmachine/tests/test_labelled_spatial_aggregate.py

```diff
@@ -28,6 +28,41 @@ def test_labelled_spatial_aggregate(get_dataframe):
     assert len(df) == 50
     assert len(df.pcod.unique()) == 25
     assert len(df.label_value.unique()) == 2
+    # Total number of values should equal the initial number of subscribers
+    assert sum(map(int, df.value.tolist())) == len(get_dataframe(metric))
```

Comment:
I think it would make more sense to compare to `len(get_dataframe(locations))` here - in the event that the set of subscribers differed between `locations` and `metric`, we'd still expect each row in the `locations` query to be counted once.

Also, is there a reason not to use `df.value.sum()` for the left-hand side of this check?

Reply:
Could also consider testing that `df.groupby("pcod").value.sum()` is the same as the result of `SpatialAggregate(locations=locations)` (i.e. counts per location sum to the correct values).